Modules/_sqlite/connection.c

```diff
@@ -1461,13 +1451,13 @@ pysqlite_collation_callback(
         int text1_length, const void* text1_data,
         int text2_length, const void* text2_data)
 {
+    PyGILState_STATE gilstate = PyGILState_Ensure();
+
     PyObject* string1 = 0;
     PyObject* string2 = 0;
-    PyGILState_STATE gilstate;
     PyObject* retval = NULL;
     long longval;
     int result = 0;
-    gilstate = PyGILState_Ensure();
 
     if (PyErr_Occurred()) {
```

Comment:
Digression: is this really needed? If `PyGILState_Ensure()` fails, won't we just crash intensely anyway?

Reply:
`PyErr_Occurred` is Python API and must be called with the GIL held.
The comment is a good idea though :)

Reply:
> PyErr_Occurred is Python API and must be called with the GIL held.

Doh, of course 🤦🏻 😄  I'll add a comment right away.

Reply:
See 2a2152dd40bbf78c4d160a98720b264c1fc60cdf